src/librustc/ty/sty.rs

```diff
@@ -2276,6 +2230,10 @@ impl<'tcx> Const<'tcx> {
                     }
                 }
             }
+            ConstValue::Unevaluated(..) => {
+                flags |= TypeFlags::HAS_NORMALIZABLE_PROJECTION;
+                flags |= TypeFlags::HAS_PROJECTION;
```

Comment:
You need to include flags from the substs, I think.

Reply:
Basically treat `Unevaluated` like `ty::{Adt,Projection,UnnormalizedProjection}` - unless they differ on something, in which case it needs to be looked at in more detail.

Reply:
those are included via the `add_const` method that calls this method. All of that logic is slightly weirdly distributed. Not sure what's going on there. We should refactor these somewhat, but that's a preexisting issue.

Reply:
Okay I'm confused as to why this `Const::type_flags` method even exists, I'll look into it a bit.

Reply:
Ah looks like this is a mistake from the const generics work, mimicking `RegionKind` (which idk why it has a `type_flags` method but it's not great).

This should be inlined/moved into `add_const`.

Reply:
opened https://github.com/rust-lang/rust/issues/59209 to track it

Reply:
Oh I wanted you to do it here since it should be pretty easy (only one call site) and so I can actually review it. I can't tell very well if this is correct because it's awkwardly split into two.